Baskerville/llama2-inference-sbatch.md

```diff
@@ -0,0 +1,433 @@
+# Llama 2 inference on Baskerville using `sbatch`
+
+Before you can make use of them you’ll need to download the models.
+See [llama-download.md](llama-download.md) for how to go about doing this on Baskerville.
+
+## Llama 2 7B inference, single node, single GPU
+
+The 7B parameter Llama 2 model will run with a single process on a single node with a single GPU.
+The checkpoint provided for use with the example will, in fact, only work with a single process.
+
+We found at least 16 GiB of RAM was needed to run the model.
+With too little RAM the process will be sent a SIGKILL termination signal, the process will end with an exit code of -9 and an OOM report will be generated.
+
+The RAM, node, GPU and process quantities must be explicitly configured through `sbatch` header for things to work.
+
+## 1. Get the full path of your llama directory.
+
+You should already have the llama git repository cloned to Baskerville.
+You’ll need the full path of where you cloned it to for the script.
+You can find this by moving into the directory and then running the `pwd` (print working directory) command:
+
+```shell
+$ cd llama
+$ pwd
+```
+
+## 2. Create the batch script.
+
+Using your favourite text editor, create a batch file with the contents below.
+
+In this script you’ll need to replace `<PROJECT_ID>` with the name of your project and `<LLAMA2_PATH>` with the full path of your cloned llama repository from Step 1.
+
+Make sure you call the script `batch-llama2-7b-inf.sh`.
+Technically you can give it any name, but I’ve assumed it’s called this in the steps that follow.
+
+```shell
+#!/bin/bash
+#SBATCH --qos turing
+#SBATCH --account <PROJECT_ID>
+#SBATCH --time 0:30:0
+#SBATCH --nodes 1
+#SBATCH --gpus 1
+#SBATCH --mem 16384
+#SBATCH --job-name llama2-7b-inf
+
+# Execute using:
+# sbatch -o stdout-%A_%a.out ./batch-llama2-7b-inf.sh
+
+module purge
+module load baskerville
+module load bask-apps/live
+module load PyTorch/2.1.2-foss-2022b-CUDA-11.8.0
+
+VENV_PATH="./venv"
+
+pushd <LLAMA2_PATH>
+
+if [ -d "$VENV_PATH" ]; then
```

Comment:
Will VENV_PATH be relative to the LLAMA2_PATH and have we mentioned that users might want to make a virtual environment in their Llama2 folder / is it done automatically by the download script?

Reply:
Yes, exactly. The default value for `VENV_PATH` is `./venv` and this will be relative to whatever the user replaces `<LLAMA2_PATH>` with. If there's already a virtual environment there with this name it'll be used, but otherwise one will be created automatically by the script.

I've added some text in above the script to explain this. Will this make things clearer?

> If it doesn't already exist, a virtual environment will be created and configured automatically in a directory called `venv` immediately inside the directory you set `<LLAMA2_PATH>` to.